### DISCUSSION TITLE
Examples not working with SQLx 0.6.0

### DISCUSSION THREAD

**Discussion by pintariching** [rwf2/Rocket#2241] [Questions] [answered]
I get the error 
```
.fetch_all(&mut *db)
--------- ^^^^^^^^ the trait `sqlx::Executor<'_>` is not implemented for `&mut rocket_db_pools::sqlx::pool::PoolConnection<rocket_db_pools::sqlx::Postgres>
```

when I try to follow the example code for using Rocket with SQLx 0.6.0 and rocket_db_pools
It compiles when using SQLx 0.5.13 so I'm guessing some traits need to be implemented?
Or do I have to pass the connection pool in another way? Right now I'm doing it like this:
```
#[get("/tags")]
pub async fn fetch(
	mut db: Connection<Db>
) -> Result<Json<Vec<Tag>>, (Status, Json<ApiError>)> {
	sqlx::query_as!(Tag, "SELECT id, label, color FROM tag ORDER BY id")
		.fetch_all(&mut *db)
		.await
		.map(|tag| Json(tag))
		.map_err(|e| {
			ApiError::server_error(
				Status::InternalServerError, 
				&e.to_string()
			)
		})
}
```

**Comment by crystal-patronus** [comment]
I also have same problem. Did you fix it?

**Comment by the10thWiz** [comment]
Do you have `sqlx` listed in your crate's Cargo.toml? You shouldn't need to, since Rocket re-exports the entire crate, and it's guaranteed to be the same version that is used internally.

**Comment by crystal-patronus** [comment]
I already solved it using dereference.
Thank you.

**Comment by lammas** [comment, marked as answer]
Just so it's absolutely clear what worked: `.fetch_all(&mut *db)` became `.fetch_all(&mut **db)`.
